iwyu.cc

```diff
@@ -3842,10 +3865,8 @@ class InstantiatedTemplateVisitor
     // CanForwardDeclareType function relies on the specific placement of
     // the type node in the AST. An intermediate SubstTemplateTypeParmType could
     // break that logic. However, such cases don't even need to be considered
-    // here, because they are handled in VisitSubstTemplateTypeParmType. But
-    // maybe using HasAncestorOfType, or replacing VisitSubst...Type with
-    // TraverseSubst...Type and Traverse...TypeLoc would be more reliable.
-    if (!current_ast_node()->ParentIsA<SubstTemplateTypeParmType>())
+    // here, because they are handled in VisitSubstTemplateTypeParmType.
+    if (!current_ast_node()->HasAncestorOfType<SubstTemplateTypeParmType>())
```

Comment:
An alternative could be `MostElaboratedAncestor`.

Reply:
Yeah, let's do that if we know it's only elaborated types getting in the way.